src/dfx/src/commands/language_service.rs

```diff
@@ -40,7 +44,38 @@ pub fn exec(env: &dyn Environment, opts: LanguageServiceOpts) -> DfxResult {
             .get_defaults()
             .get_build()
             .get_packtool();
-        let package_arguments = package_arguments::load(env.get_cache().as_ref(), packtool)?;
+
+        let mut package_arguments = package_arguments::load(env.get_cache().as_ref(), packtool)?;
+
+        // Include actor alias flags
+        let canister_names = config
+            .get_config()
+            .get_canister_names_with_dependencies(None)?;
+        let network_descriptor = get_network_descriptor(
+            env.get_config(),
+            None, /* opts.network */
+            LocalBindDetermination::ApplyRunningWebserverPort,
+        )?;
+        let canister_id_store =
+            CanisterIdStore::new(&network_descriptor, env.get_config(), env.get_temp_dir())?;
```

Comment:
Is this what you were imagining for `CanisterIdStore::new()`? Besides still having to call `get_network_descriptor()` because of the non-AgentEnvironment, everything else seems cleaner. 

Also, the language server is no longer resolving canister ids as of merging #2376 (with or without the AgentEnvironment). Everything still works as expected on the master branch. Do you have a guess to what could be causing the empty CanisterIdStore? (@ericswanson-dfinity)

Reply:
Yes, this is what I expected it would look like.  At first glance, I don't see why canister id lookup would be failing.